lib/idempo.rb

```diff
@@ -127,10 +124,15 @@ def response_may_be_persisted?(status, headers, body)
   end
 
   def body_size_within_limit?(response_headers, body)
-    return response_headers["Content-Length"].to_i <= SAVED_RESPONSE_BODY_SIZE_LIMIT if response_headers["Content-Length"]
+    lowercase_response_headers = response_headers.map do |(k, v)|
+      [k.downcase, v]
+    end.to_h
 
-    return false unless body.is_a?(Array) # Arbitrary iterable of unknown size
+    if response_headers["content-length"]
```

Comment:
Is this meant to check `lowercase_response_headers` as well? 🧐

Reply:
Yes, I will do another pass to ensure all headers are lowercase everywhere. Just a few more steps.